app/models/concerns/indexable.rb

```diff
@@ -30,6 +30,21 @@ def create_index
         body:  { settings: settings, mappings: mappings })
     end
 
+    def update_metadata(version)
+      ES.client.index(
+        index: index_name,
+        type: 'metadata',
+        id: 0,
+        body: { version: version, time: DateTime.now.utc })
+    end
+
+    def stored_metadata
+      ES.client.search(
+        index: index_name,
+        type: 'metadata',
+        body: {query: {match: {_id: 0}}})['hits']['hits'][0]['_source'].symbolize_keys rescue {}
```

Comment:
Yeah I guess what I want is something like short-circuit evaluation on a chain of method calls: `['hits']['hits'][0]['_source']` -- instead of `if/else` checking for all of those keys, just let me give another expression for when one of them is absent.
I wish you could do `{}[0][0] rescue(NoMethodError) {}`.
I also wish you could just catch exceptions thrown in the immediate execution context: `ES.client.search(...)[:some][:keys] rescue_local(NoMethodError) {}` but I guess that's really being thrown in an instance of NilClass...
    class << nil
      def [](*_); self; end
    end
Hmm... screw it.